exporter/elasticsearchexporter/data_stream_router.go

```diff
@@ -187,6 +195,9 @@ func routeRecord(
 		if submatch := receiverRegex.FindStringSubmatch(scope.Name()); len(submatch) > 0 {
 			receiverName := submatch[1]
 			dataset = receiverName
+		} else if selfTelemetryScopeNames[scope.Name()] {
+			// For collector self-telemetry, use a fixed dataset name
+			dataset = collectorSelfTelemetryDataStreamDataset
 		}
```

Comment:
nit: is it easier to do self telemetry exact match before the regex match? i.e. swapping the `if` and `else if`. Then we can avoid other regex bugs down the line, or even skip the regex change in this PR.